packages/react-native-renderer/src/ReactFabricHostConfig.js

```diff
@@ -193,6 +217,101 @@ class ReactFabricHostComponent {
 
     return;
   }
+
+  // This API (dispatchEvent, addEventListener, removeEventListener) attempts to adhere to the
+  // w3 Level2 Events spec as much as possible, treating HostComponent as a DOM node.
+  //
+  // Unless otherwise noted, these methods should "just work" and adhere to the W3 specs.
+  // If they deviate in a way that is not explicitly noted here, you've found a bug!
+  //
+  // See:
+  // * https://www.w3.org/TR/DOM-Level-2-Events/events.html
+  // * https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/dispatchEvent
+  // * https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/addEventListener
+  // * https://developer.mozilla.org/en-US/docs/Web/API/EventTarget/removeEventListener
+  dispatchEvent_unstable(event: CustomEvent) {
```

Comment:
Originally we designed this object to be completely removed so that we could reclaim the memory and to optimize calls by removing virtual dispatch. Eg by introducing apis like focus(ref) instead.

These other methods were just there for backwards compatibility.

That was never followed through and so this was never deleted.

So this direction seems like a reversal of that strategy and to keep living with more memory usage and worse performance. If that’s the direction then there are probably more things that should be cleaned up.

Reply:
> this direction seems like a reversal of that strategy

The other option was to do `addEventListener(hostElement)`